README.md

```diff
@@ -0,0 +1,284 @@
+# DNS=8.8.8.8
+```
+
+## Detailed Description
+
+* A key `x` in `systemd_network_netdevs` causes an INI file
+  `/etc/systemd/network/x.netdev` to be created.
+  By default it contains:
+
+  ```ini
+  [NetDev]
+  Name=x
+  ```
+
+  Under `systemd_network_netdevs['x']`,
+  the first-layer keys turn into INI sections,
+  the second-layer keys turn into INI properties and
+  the values thereunder turn into the respective INI values.
+
+* A key `x` in `systemd_network_networks` causes an INI file
+  `/etc/systemd/network/x.network` to be created.
+  By default it contains:
+
+  ```ini
+  [Match]
+  Name=x
+  ```
+
+  `systemd_network_networks['x']` analogously turns into INI, with one
+  special behaviour:
+  If you specify a `Match` key (turning into a `[Match]` section) inside
+  `systemd_network_networks['x']`, then the default `Name=x` **does not apply**.
+  This makes it possible to
+  [match on different things than name](https://www.freedesktop.org/software/systemd/man/systemd.network.html#%5BMatch%5D%20Section%20Options).
+
+If you need multiple sections (respectively properties) with the same
+name, then specify that section (respectively property) once and give
+a list at the next YAML layer.
+Examples:
+
+```yml
+# Example for multiple [Route] sections
+systemd_network_networks:
+  enp1s0:
+    Network:
+      Address: 192.168.1.2/24
+    Route:
+      - Destination: 192.168.5.0/24
+        Gateway: 192.168.1.1
+      - Destination: 192.168.6.0/24
+        Gateway: 192.168.1.1
+
+##### Turns into:
+#
+##### /etc/systemd/network/enp1s0.network
+#
+# [Match]
+# Name=enp1s0
+#
+# [Network]
+# Address=192.168.1.2/24
+#
+# [Route]
+# Destination=192.168.5.0/24
+# Gateway=192.168.1.1
+#
+# [Route]
+# Destination=192.168.6.0/24
+# Gateway=192.168.1.1
+```
+
+```yml
+# Example for multiple VLAN= properties
+systemd_network_networks:
+  enp1s0:
+    Network:
+      VLAN:
+        - enp1s0.10
+        - enp1s0.11
+
+##### Turns into:
+#
+##### /etc/systemd/network/enp1s0.network
+#
+# [Match]
+# Name=enp1s0
+#
+# [Network]
+# VLAN=enp1s0.10
+# VLAN=enp1s0.11
+```
+
+## DNS resolver
+
+This role automatically disables resolvconf, enables systemd-resolved
+and configures the `/etc/resolv.conf` symlink correctly for systemd-resolved.
+Using systemd-resolved (as this role does by default) is recommended because
+systemd-resolved reads the DNS servers from your systemd network configuration,
+whereas the DNS servers for resolvconf can't be configured using this role.
+
+If you want to use resolvconf instead, there's a role variable for that:
+
+```yml
+systemd_network_dns_resolver: resolvconf
+```
+
+## Upload extra files
+
+There also is a role variable `systemd_network_copy_files` which
+simply takes a list of which each element is passed to the Ansible
+[copy module](https://docs.ansible.com/ansible/latest/modules/copy_module.html).
+
+### WireGuard client example
+
+`systemd_network_copy_files` can for example be used for WireGuard key
```

Comment:
Why not implement this functionality in the wireguard role?

Reply:
The wireguard role just installs WireGuard. WireGuard is configured using the systemd-network role.